tests/lib/fde-setup-hook/fde-setup.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+// super secure crypto
+func xor13(bs []byte) []byte {
+	out := make([]byte, len(bs))
+	for i := range bs {
+		out[i] = bs[i] ^ 0x13
+	}
+	return out
+}
+
+// XXX: import struct from snapd instead?
+type fdeSetupJSON struct {
+	Op string `json:"op"`
+
+	Key     []byte `json:"key,omitempty"`
+	KeyName string `json:"key-name,omitempty"`
+
+	Models []map[string]string `json:"models,omitempty"`
+}
+
+func runFdeSetup() error {
+	output, err := exec.Command("snapctl", "fde-setup-request").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("cannot run snapctl fde-setup-request: %v", osutil.OutputErr(output, err))
+	}
+	var js fdeSetupJSON
+	if err := json.Unmarshal(output, &js); err != nil {
+		return err
+	}
+
+	var fdeSetupResult []byte
+	switch js.Op {
+	case "features":
+		// no special features supported by this hook
+		fdeSetupResult = []byte("[]")
```

Comment:
With https://github.com/snapcore/snapd/pull/9791 this needs to change, but we have options